### ISSUE TITLE
Document+improve the `/etc/sudoers` situation

### ISSUE THREAD

**Issue by LukeShu** [telepresenceio/telepresence#2314]
**Please describe your use case / problem.**

It is a not-infrequent ask to be able to run Telepresence without it prompting you for your password.  While this is possible to configure, it is (1) undocumented, and (2) what you have to do isn't great.

When Flynn has asked me how to do this in the past, I have told him the simple and obvious thing: to stick this in his `/etc/sudoers`:
```sudoers
# THIS CONFIG DOESN'T WORK, DO NOT USE
# SUDO_USERS HOSTS=(AS_USER) TAGS COMMANDS
flynn ALL=(ALL) NOPASSWD: /Users/flynn/src/github.com/datawire/ambassador/tools/bin/telepresence
```
Unfortunately, that doesn't actually work (see below).

He's followed that up by asking if he should just make the executable SUID root.  This would be bad; several parts of Telepresence really want to run as the normal user, and making it SUID root would make that impossible.  (And also, marking a binary as SUID that wasn't specifically designed to be marked as SUID is generally a bad idea, but lotsa people do it anyway.)  If Flynn thought about doing this, lotsa outside users probably have as well, so that's a bad thing.

OK, so I (lukeshu) have prior knowledge know that it does a preflight check with `sudo true`, so I tried adding `/usr/bin/true` to that line:
```sudoers
# THIS CONFIG DOESN'T WORK, DO NOT USE
# SUDO_USERS HOSTS=(AS_USER) TAGS COMMANDS
lukeshu ALL=(ALL) NOPASSWD: /home/lukeshu/src/github.com/datawire/ambassador/tools/bin/telepresence, /usr/bin/true
```
Well, this doesn't work either:
```console
$ telepresence connect
telepresence: error: daemon service did not start (see "/home/lukeshu/.cache/telepresence/logs/daemon.log" for more info)
failed to scan connector logs: open /home/lukeshu/.cache/telepresence/logs/connector.log: no such file or directory
```
extra mysterious is that `daemon.log` is empty! A "normal" user would have no idea how to debug this.  This isn't a good failure mode!

So, I consult `pkg/proc/exec_unix.go`, and try stepping through the steps that it's doing.  And then I find the culprit:
```console
$ sudo --non-interactive --preserve-env ./tools/bin/telepresence 
sudo: sorry, you are not allowed to preserve the environment
```

So then I consulted `man sudoers` and then changed the configuration to
```sudoers
# This config actually works
# SUDO_USERS HOSTS=(AS_USER) TAGS COMMANDS
lukeshu ALL=(ALL) NOPASSWD: SETENV: /home/lukeshu/src/github.com/datawire/ambassador/tools/bin/telepresence
lukeshu ALL=(ALL) NOPASSWD: /usr/bin/true
```

(PS: allowing `SETENV:` for `true` would be bad because an attacker could set `LD_PRELOAD` to turn it in to a generic privilege escalation. Telepresence is immune to this attack because it is statically linked; though other attacks may be possible.)

OK, so what all problems have we seen?
 - None of this is documented; the only way to figure it out is by reading the source.
 - Requiring an external binary means asking for permission on that binary.
 - Just because the user can `sudo true` doesn't mean that they can `sudo --preserve-env telepresence`
 - If `sudo` fails, that doesn't logged or reported anywhere.

**Describe the solution you'd like**

Revisiting the 4 problems identified:
 - None of this is documented; the only way to figure it out is by reading the source.
   + How to do this should be part of the documentation.
 - Requiring an external binary means asking for permission on that binary.
   + A `telepresence true` or `telepresence noop` or whatever hidden/internal subcommand should be added, and it should use that for the preflight checks instead of `true`
 - Just because the user can `sudo true` doesn't mean that they can `sudo --preserve-env telepresence`
   + Either the preflight checks (with and without `--non-interactive`) should all say `--preserve-env`, or it should drop `--preserve-env` entirely and it should pass the nv-vars that it cares about as arguments.
 - If `sudo` fails, that doesn't logged or reported anywhere.
   + Figure out what needs changed to capture the output safely.  I'm thinking the best solution might be for `daemon.log` to already be opened on `sudo`'s stdout... but that'd be a surprisingly invasive change.

**Versions**
  - Output of `telepresence version`
    ```
    Root Daemon: v2.4.2 (api v3)
    User Daemon: v2.4.2 (api v3)
    ```
    (but I verified that the relevant code hasn't changed in newer versions)

**Comment by thallgren** [comment]
We're currently working on making the daemon processes long-running and installable as proper daemons. Once that's complete, this will no longer be an issue (telepresence will never start the root daemon).

With that in mind, I think getting the docs describing the current behavior out there is crucial (the real friction cause) but improving the implementation shouldn't be prioritized at this point.

**Comment by thallgren** [comment]
Closing this since docs and bug-fix are merged into the 2.5.0 release.

**Comment by Yamakaky** [comment]
Could you point me to where it is mentioned in the doc? I couldn't find it at https://www.telepresence.io/docs/latest/quick-start/...
